src/commands/CmdSync.cpp

```diff
@@ -98,9 +98,9 @@ int CmdSync::execute(std::string& output) {
         Context::getContext().config.get("sync.aws.secret_access_key");
     std::string aws_default_credentials =
         Context::getContext().config.get("sync.aws.default_credentials");
-    if (aws_region == "") {
-      throw std::string("sync.aws.region is required");
-    }
+    std::string aws_endpoint_url = Context::getContext().config.get("sync.aws.endpoint_url");
+    bool aws_force_path_style =
+        Context::getContext().config.getBoolean("sync.aws.force_path_style");
```

Comment:
What value does this need to take when it and `aws_endpoint_url` are omitted? I assume this call will default to `false` when the config is not provided.

Reply:
We'll also need to update the docs as to when `region` is required, and allow it to be empty in other cases.

Reply:
Indeed, this defaults to `false`.

I believe AWS will actually work with both `aws_force_path_style = false` and `aws_force_path = true`!

:ack: on the docs here, I need to start editing ./doc/man/task-sync.5